src/network.rs

```diff
@@ -177,13 +194,21 @@ impl Nectar {
         Ok(())
     }
 
-    fn confirm(&mut self, order: TakenOrder) {
+    fn confirm(&mut self, order: TakenOrder) -> anyhow::Result<()> {
+        self.active_takers.insert(order.taker)?;
+
         self.orderbook.confirm(order.id, order.confirmation_channel);
+
+        Ok(())
     }
 
-    fn deny(&mut self, order: TakenOrder) {
+    fn deny(&mut self, order: TakenOrder) -> anyhow::Result<()> {
+        self.active_takers.remove(&order.taker)?;
```

Comment:
❓ Is this still needed. With the logic of `active_takers` being moved to the network layer  there should not be an error that leads to the a taker being stored but then running into deny. (It does not harm us to do it at this point, thus I only raised a question.)

Reply:
You're right, we don't need this :+1:

Reply:
I'll fix it with PR #62.

Reply:
Nice catch!